homeassistant/components/atome/sensor.py

```diff
@@ -167,25 +157,43 @@ def _retrieve_period_usage(self, period_type):
             _LOGGER.error(
                 "%s : Missing last value in values: %s: %s", period_type, values, error
             )
+        retrieve_has_been_sucessful = not error_during_retrieve
+        return retrieve_has_been_sucessful, period_usage, period_price
```

Comment:
Is it a problem for period_usage to be updated if period_price fails?

Reply:
Normally server shall send back both

Reply:
I ve put `if` statement with both condition